src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -54,65 +44,57 @@ public NativeInterface(AndroidJavaObject config)
       using (var unityPlayerClass = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
       using (var activity = unityPlayerClass.GetStatic<AndroidJavaObject>("currentActivity"))
       using (var context = activity.Call<AndroidJavaObject>("getApplicationContext"))
+      using (var client = new AndroidJavaObject("com.bugsnag.android.Client", context, config))
       {
         // lookup the NativeInterface class and set the client to the local object.
         // all subsequent communication should go through the NativeInterface.
-        var client = new AndroidJavaObject("com.bugsnag.android.Client", context, config);
         var nativeInterfaceRef = AndroidJNI.FindClass("com/bugsnag/android/NativeInterface");
         BugsnagNativeInterface = AndroidJNI.NewGlobalRef(nativeInterfaceRef);
         AndroidJNI.DeleteLocalRef(nativeInterfaceRef);
         var setClient = AndroidJNI.GetStaticMethodID(BugsnagNativeInterface, "setClient", "(Lcom/bugsnag/android/Client;)V");
 
-        jvalue[] args = AndroidJNIHelper.CreateJNIArgArray(new object[] {client});
+        var argsArray = new object[] {client};
+        jvalue[] args = AndroidJNIHelper.CreateJNIArgArray(argsArray);
         AndroidJNI.CallStaticVoidMethod(BugsnagNativeInterface, setClient, args);
+        AndroidJNIHelper.DeleteJNIArgArray(argsArray, args);
 
         // Cache JNI refs which will be used to load report data later in the
         // app lifecycle to avoid repeated lookups
-        var unityRef = AndroidJNI.FindClass("com/bugsnag/android/unity/BugsnagUnity");
-        BugsnagUnityClass = AndroidJNI.NewGlobalRef(unityRef);
```

Comment:
My understanding is that this is required even though it looks like dead code, because it loads the `BugsnagUnity` class which in turn invokes `System.loadLibrary`.